libcalico-go/lib/clientv3/ippool.go

```diff
@@ -515,6 +527,25 @@ func (r ipPools) UnsafeCreate(ctx context.Context, res *apiv3.IPPool, opts optio
 	return nil, err
 }
 
+func (r ipPools) DebugUnsafeCreateNoDefaults(ctx context.Context, res *apiv3.IPPool, opts options.SetOptions) (*apiv3.IPPool, error) {
```

Comment:
Hm, I'm not sure about adding a new public method to the interface just to satisfy this test.

We have other ways of writing "raw" data into the datastore - e.g., using a backend client directly. 

It's easy to get a backend client from a client - we do it in some of our other tests as well, like here: https://github.com/projectcalico/calico/blob/bc619e7ef607229f40f473c8baf60096a2eb58ab/kube-controllers/pkg/controllers/node/kdd_ipam_gc_fv_test.go#L97-L101

Reply:
You can then use this function directly: https://github.com/projectcalico/calico/blob/bc619e7ef607229f40f473c8baf60096a2eb58ab/libcalico-go/lib/backend/api/api.go#L62-L65

e.g.,

```go
kvp := model.KVPair{
  Key: model.ResourceKey{},
  Value: &v3.IPPool{},
}
bc.Create(ctx, kvp)
```

Reply:
I see, yes this is much better. Will do, thanks!